dashboard/app/models/level_source_image.rb

```diff
@@ -71,7 +66,22 @@ def upload_framed_image(image)
     upload_image(s3_framed_filename, framed_image)
   end
 
-  def self.hashify_filename(plain)
+  def delete_image_or_framed_image
+    AWS::S3.delete_from_bucket(S3_BUCKET, s3_filename)
+    AWS::S3.delete_from_bucket(S3_BUCKET, s3_framed_filename)
+  end
+
+  private
+
+  # @return [String] The filename for the image frame.
+  def framed_image_filename
```

Comment:
I believe this will need to be called `frame_image_filename` if you intend for it to be a drop-in replacement for the removed if/else in `upload_framed_image`